shiny/express/_recall_context.py

```diff
@@ -54,6 +54,19 @@ def __exit__(
             sys.displayhook(res)
         return False
 
+    def tagify(self) -> Tag | TagList | MetadataNode | str:
+        res = self.fn(*self.args, **self.kwargs)
+
+        if callable(getattr(res, "tagify", None)):
+            return res.tagify()  # pyright: ignore
+        if callable(getattr(res, "_repr_html_", None)):
+            return res._repr_html_()  # pyright: ignore
+
+        raise RuntimeError(
+            "RecallContextManager was used without `with`. When used this way, the "
+            "result must have a .tagify() or ._repr_html_() method, but it does not."
+        )
+
```

Comment:
Note this `tagify` method on `RecallContextManager`.

Reply:
Is there a reason this is stricter than `__exit__`? Oh, is it because sys.displayhook will do something similar?

Reply:
Never mind, I understand